roles/collect/tasks/main.yml

```diff
@@ -37,39 +37,28 @@
   when:
     - not check_oc.stat.exists
 
-- name: Default OCP insecure login option
+- name: Set facts from OCP config
   set_fact:
-    ocp_insecure_login: '--insecure-skip-tls-verify=false'
+    ocp_validate_cert: '{{ ocp_config["ocp_validate_cert"] }}'
+    ocp_metering_namespace: '{{ ocp_config["ocp_metering_namespace"] }}'
 
-- name: Set OCP insecure login option
+- name: Set facts from OCP config
   set_fact:
-    ocp_insecure_login: '--insecure-skip-tls-verify=true'
-  when:
-    - 'not {{ ocp_config["ocp_validate_cert"] | bool }}'
-
-- name: Define OCP API endpoint without proxy
-  set_fact:
-    collect_endpoint_path: 'api/v1/namespaces/{{ ocp_config["ocp_metering_namespace"] }}'
-  when: not collect_with_proxy
+    ocp_insecure_login: '--insecure-skip-tls-verify={{ ocp_validate_cert }}'
+    collect_endpoint_params: '?name={{ collect_report_name }}&format={{ collect_format }}&namespace={{ ocp_metering_namespace }}'
+    collect_endpoint_path: 'api/v1/namespaces/{{ ocp_metering_namespace }}'
+    collect_endpoint_url: '{{ ocp_config["ocp_api"] }}'
 
-- name: Define OCP API endpoint with proxy
+# conditional tag
+- name: Define facts from OCP config with proxy
   set_fact:
-    collect_endpoint_path: 'api/v1/namespaces/{{ ocp_config["ocp_metering_namespace"] }}/services/https:reporting-operator:http/proxy/api/v1/scheduledreports/get'
-  when: collect_with_proxy
+    collect_endpoint_path: '{{ collect_endpoint_path }}/services/https:reporting-operator:http/proxy/api/v1/scheduledreports/get'
+    collect_endpoint_url: '{{ collect_proxy_server_no_port }}:{{ ocp_config["ocp_proxy_port"] }}'
+  tags: with_proxy
 
 - name: Define OCP API URL without proxy
   set_fact:
-    collect_endpoint_url: '{{ ocp_config["ocp_api"] }}/{{ collect_endpoint_path }}/{{ collect_endpoint_params }}'
-  when: not collect_with_proxy
-
-- name: Define OCP API URL with proxy
-  set_fact:
-    collect_endpoint_url: '{{ collect_proxy_server_no_port }}:{{ ocp_config["ocp_proxy_port"] }}/{{ collect_endpoint_path }}{{ collect_endpoint_params }}'
-  when: collect_with_proxy
-
-- name: Set collection uuid
-  set_fact:
-    current_csv_uuid: '{{ collect_csv_uuid }}'
```

Comment:
This was added as a bug fix to a previous issue found running korekuta where each time `collect_csv_uuid` was used within the role a different uuid. It was acting like a function call almost, could have been because the manifest template is also Jinja2, but the value between the path and in the manifest file were different.